src/groupByKey.ts

```diff
@@ -1,7 +1,26 @@
+import { error } from 'console';
+
 type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> | null {
+  const result: GroupsMap<T> = {};
+
+  for (const item of items) {
+    const name = (item[key] as string) || null;
+
+    if (name) {
+      if (result[name]) {
+        continue;
+      }
+      result[name] = items.filter((data) => data[key] === name);
+    } else {
+      error('check typeof key in T');
+
+      return null;
```

Comment:
Returning `null` when the key is not a string might not be the best approach. Consider throwing an error or handling the situation in a way that provides more information to the caller.